### PR TITLE
fix: added start and end time info text to educate user better around the schedule timelines

### PR DESCRIPTION
**Pull request by SagarRajput-7**
### Summary

```
- Does not repeat
    - StartTime 
      - Schedule will start at 22:00 on 1st Aug 2024
    - EndTime
      - Schedule will end at 22:00 on 2nd Aug 2024

- Recurrence -
    - StartTime
    (daily)
      - Schedule will start from 1st Aug 2024 and will trigger at 22:00 daily
    (monthly)
      - Schedule will start from 6th Aug 2024 and will trigger at 22:00 on 6th day of every month
    (weekly)
      - Schedule will start from 6th Aug 2024 and will trigger at 22:00 on [Monday, Wednesday] every week.

    - EndTime
      - Schedule will end at 22:00 on 2nd Aug 2024
 ```

#### Related Issues / PR's

https://github.com/SigNoz/engineering-pod/issues/1712

#### Screenshots

- Fixed schedule:

image

---------------

- Recurrence - daily - 

image

----------------

- Recurrence - monthly - 

image

-----------------
- Recurrence - weekly - 

image

----------------

Video - 

https://github.com/user-attachments/assets/9af63810-67be-4e13-887a-320a4a5283ba

----------------

### CODE REVIEW

File: frontend/src/container/PlannedDowntime/PlannedDowntimeForm.tsx [hunk 1]
```diff
@@ -300,6 +304,111 @@ export function PlannedDowntimeForm(
...
+			formData.timezone,
+			!isEditMode,
+			'HH:mm',
+		);
+
+		const formattedEndDate = timezoneFormatted(
+			endTime,
+			formData.timezone,
+			!isEditMode,
+			'Do MMM YYYY',
```
Comment (rahulkeswani101) [inline frontend/src/container/PlannedDowntime/PlannedDowntimeForm.tsx:407]: Can you please create some constants for the date-time format

**Comment by srikanthccv** [comment]
@SagarRajput-7 looks good to me. Just some wording suggestions.

## start

fixed:

Scheduled for September 5, 2024 starting at 13:36

daily:

Scheduled from September 5, 2024, daily starting at 13:36.

weekly:

Scheduled from September 5, 2024, weekly on Mondays and Wednesdays starting at 13:36

monthly:

Scheduled from September 5, 2024, monthly on the 5th starting at 13:36


## end

Scheduled to end maintenance on September 27, 2024 at 13:37.

File: frontend/src/container/PlannedDowntime/PlannedDowntimeForm.tsx (hunk 1 above)
Comment (SagarRajput-7) [reply frontend/src/container/PlannedDowntime/PlannedDowntimeForm.tsx]: done

File: frontend/src/container/PlannedDowntime/PlannedDowntimeForm.tsx [hunk 2]
```diff
@@ -348,6 +467,7 @@ export function PlannedDowntimeForm(
 						popupClassName="datePicker"
 					/>
 				</Form.Item>
+				<div className="scheduleTimeInfoText">{startTimeText()}</div>
```
Comment (vikrantgupta25) [inline frontend/src/container/PlannedDowntime/PlannedDowntimeForm.tsx:470]: why this needs to be a function ? can we make a constant and memoize the same ?

File: frontend/src/container/PlannedDowntime/PlannedDowntimeForm.tsx [hunk 3]
```diff
@@ -426,6 +546,7 @@ export function PlannedDowntimeForm(
 						popupClassName="datePicker"
 					/>
 				</Form.Item>
+				<div className="scheduleTimeInfoText">{endTimeText()}</div>
```
Comment (vikrantgupta25) [inline frontend/src/container/PlannedDowntime/PlannedDowntimeForm.tsx:549]: same as above

File: frontend/src/container/PlannedDowntime/PlannedDowntimeForm.tsx [hunk 4]
```diff
@@ -300,6 +308,116 @@ export function PlannedDowntimeForm(
...
+				endTime = formData?.endTime || '';
+			}
+		}
+
+		if (!endTime) {
+			return '';
+		}
+
+		if (formData.timezone) {
+			endTime = handleTimeConvertion(
```
Comment (vikrantgupta25) [inline frontend/src/container/PlannedDowntime/PlannedDowntimeForm.tsx:397]: NIT: `handleTimeConversion` spell check

File: frontend/src/container/PlannedDowntime/PlannedDowntime.styles.scss [hunk 5]
```diff
@@ -77,6 +77,18 @@
...
+	.formItemWithBullet {
+		margin-bottom: 0;
+	}
+
+	.scheduleTimeInfoText {
+		margin-top: 8px;
+		margin-bottom: 20px;
+		font-size: 12px;
+		font-weight: 400;
+		color: var(--bg-vanilla-400);
```
[the remark this answers is not in the document]
Comment (SagarRajput-7) [reply frontend/src/container/PlannedDowntime/PlannedDowntime.styles.scss:90]: added

File: frontend/src/container/PlannedDowntime/PlannedDowntimeForm.tsx (hunk 2 above)
Comment (SagarRajput-7) [reply frontend/src/container/PlannedDowntime/PlannedDowntimeForm.tsx]: done

File: frontend/src/container/PlannedDowntime/PlannedDowntimeForm.tsx (hunk 3 above)
Comment (SagarRajput-7) [reply frontend/src/container/PlannedDowntime/PlannedDowntimeForm.tsx]: done

File: frontend/src/container/PlannedDowntime/PlannedDowntimeForm.tsx (hunk 4 above)
Comment (SagarRajput-7) [reply frontend/src/container/PlannedDowntime/PlannedDowntimeForm.tsx]: thanks, done

File: frontend/src/container/PlannedDowntime/PlannedDowntimeForm.tsx [hunk 6]
```diff
@@ -300,6 +308,116 @@ export function PlannedDowntimeForm(
 		}),
 	);
 
+	const timezoneFormatted = (
```
[the remark this answers is not in the document]
Comment (SagarRajput-7) [reply frontend/src/container/PlannedDowntime/PlannedDowntimeForm.tsx]: sure

**Review by vikrantgupta25 — APPROVED** [review]
lgtm

**Review by rahulkeswani101 — APPROVED** [review]

**Review by YounixM — APPROVED** [review]

**Pull request merged**